flake.nix

```diff
@@ -1,55 +1,67 @@
 {
   description = "Generate swift and kotlin types from haskell types";
 
-  outputs = { self, nixpkgs }:
-    let
-      systems = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ];
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/master";
```

Comment:
IMO i would get rid of `/master`; this is implicit, and stating it explicitly is kind of unintuitive because you'd expect it to pull form the latest `master` every time.